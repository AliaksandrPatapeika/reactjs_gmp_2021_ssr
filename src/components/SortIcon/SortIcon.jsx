import PropTypes from 'prop-types';
import React from 'react';

const SortIcon = ({sortOrder}) => (sortOrder === 'asc'
		  ? <i className="fa fa-sort-amount-asc" aria-hidden="true" />
		  : <i className="fa fa-sort-amount-desc" aria-hidden="true" />);

SortIcon.propTypes = {
  sortOrder: PropTypes.string.isRequired
};

export default SortIcon;
