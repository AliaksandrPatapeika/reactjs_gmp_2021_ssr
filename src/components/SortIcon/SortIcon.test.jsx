import React from 'react';
import renderer from 'react-test-renderer';

import SortIcon from './SortIcon';

describe('SortIcon', () => {
  test('renders correctly', () => {
    const sortIcon = renderer
      .create(<SortIcon sortOrder="desc" />)
      .toJSON();

    expect(sortIcon).toMatchSnapshot();
  });
});
