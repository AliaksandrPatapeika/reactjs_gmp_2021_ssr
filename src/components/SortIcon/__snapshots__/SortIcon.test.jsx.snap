// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortIcon renders correctly 1`] = `
<i
  aria-hidden="true"
  className="fa fa-sort-amount-desc"
/>
`;
